### PR TITLE
MC time allocation refactor.

### PR DESCRIPTION
**Pull request by MrsTonedOne**
## Mid-sized MC refactor:

The background subsystems now strictly obey the internal_tick_usage detected by extools.
The mc nolonger attempts to make up for spare leftover game tick by re-running subsystems that paused that tick.
MC reserve lowered to 60% of a tick.
Other tick limit defines will obey the internal_tick_usage metric.
10% of a tick is required to resume a task, otherwise its not worth it. This scales to TICK_LIMIT_RUNNING (ie, if byond wanted 50% of the tick, leaving only 50% for us, the minium runtime would calculate to 5% of a tick, not 10%), this equates to 5 milliseconds on 20fps.

(sidenote: In some situations creating lag will now make time dilation go down, because the garbage subsystem will never get allocated time to run, always yeilding to byond's internal_tick_usage requirements, its an open question on rather or not this is a feature or a bug of the new system.)

## Changelog
:cl:
del: Removed free lag. Please subscribe to the Patreon to continue receiving lag.
refactor: Background subsystems like atmos and qdel/garbage collection will now aggressively yield to byond's internal processes to avoid creating lag and time dilation. 
/:cl:

### CODE REVIEW

**Comment by Kylerace** [comment]
not my free lag!

**Comment by optimumtact** [comment]
Please consider this a friendly warning, your PR's must have a properly descriptive title and body content that explains the purpose of the PR.

Your current title and or body content does not meet requirements, so please update it.

**Comment by optimumtact** [comment]
Your PR has been edited due to having a misleading or incorrect details, consider this a friendly warning, your PR's must have a properly descriptive title and body content if they are going to do something significant.

Future violations will result in a PR close instead of edit.

File: code/__DEFINES/_tick.dm [hunk 1]
```diff
@@ -1,14 +1,19 @@
 /// Percentage of tick to leave for master controller to run
-#define MAPTICK_MC_MIN_RESERVE 70
+#define MAPTICK_MC_MIN_RESERVE 60
 /// internal_tick_usage is updated every tick by extools
-#define MAPTICK_LAST_INTERNAL_TICK_USAGE ((GLOB.internal_tick_usage / world.tick_lag) * 100)
-/// Tick limit while running normally
+#define MAPTICK_LAST_INTERNAL_TICK_USAGE ((Master.normalized_internal_tick_usage / world.tick_lag) * 100)
+/// Almount of a tick to reserve for byond regardless of its reported internal_tick_usage
```
Comment (Jared-Fogle) [inline code/__DEFINES/_tick.dm:5]:
```suggestion
/// Amount of a tick to reserve for byond regardless of its reported internal_tick_usage
```

File: code/controllers/master.dm [hunk 2]
```diff
@@ -64,6 +64,8 @@ GLOBAL_REAL(Master, /datum/controller/master) = new
 	//current tick limit, assigned before running a subsystem.
 	//used by CHECK_TICK as well so that the procs subsystems call can obey that SS's tick limits
 	var/static/current_ticklimit = TICK_LIMIT_RUNNING
+	/// normalzied version of the byond internal tick usage metric to smooth out peaks.
```
Comment (Jared-Fogle) [inline code/controllers/master.dm:67]:
```suggestion
	/// normalized version of the byond internal tick usage metric to smooth out peaks.
```

**Comment by Jared-Fogle** [comment]
whatever happened to improve not remove

**Comment by optimumtact** [comment]
Please consider this a friendly warning, your PR's must have a properly descriptive title and body content that explains the purpose of the PR.

Your current title and or body content does not meet requirements, so please update it.

Your PR will be closed if you do not update the PR title

**Pull request closed without merging**